src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly

```diff
@@ -227,6 +233,12 @@ THE SOFTWARE.
         var tr = findAncestor(e,"TR");
         e.disabled = false;
         e.setAttribute('tooltip', findAncestor(e, "TD").getAttribute('data-tooltip-enabled'));
+
+        var submitButtonPresent = document.getElementsByClassName("submit-button").length > 0;
+        if (!submitButtonPresent) { // readonly mode if there's no submit button (Jenkins 2.222+)
+            e.disabled = true;
+            return;
+        }
```

Comment:
Is this still what you would consider the preferred way to make this distinction, or would a check for `readOnlyMode` be better?

Reply:
Initially I had added a hidden input with the readOnly status in it. But you asked for it to be detected instead so I changed it to this which works well.

I wouldn’t want to use jelly vars in inline JavaScript really as it would make it more difficult to extract it